src/plugins/vis_augmenter/public/view_events_flyout/components/view_events_flyout.tsx

```diff
@@ -0,0 +1,131 @@
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiFlyout,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+} from '@elastic/eui';
+import './styles.scss';
+import { VisualizeEmbeddable } from '../../../../visualizations/public';
+import { TimeRange } from '../../../../data/common';
+import { BaseVisItem } from './base_vis_item';
+import { DateRangeItem } from './date_range_item';
+import { LoadingFlyoutBody } from './loading_flyout_body';
+import { ErrorFlyoutBody } from './error_flyout_body';
+import { EventsPanel } from './events_panel';
+import { TimelinePanel } from './timeline_panel';
+import { fetchVisEmbeddable, createEventEmbeddables, createTimelineEmbeddable } from './utils';
+import { EventVisEmbeddablesMap } from './types';
+
+interface Props {
+  onClose: () => void;
+  savedObjectId: string;
+}
+
+export const DATE_RANGE_FORMAT = 'MM/DD/YYYY HH:mm';
+
+export function ViewEventsFlyout(props: Props) {
+  const [visEmbeddable, setVisEmbeddable] = useState<VisualizeEmbeddable | undefined>(undefined);
+  // This map persists a plugin resource type -> a list of vis embeddables
+  // for each VisLayer of that type
+  const [eventVisEmbeddablesMap, setEventVisEmbeddablesMap] = useState<
+    EventVisEmbeddablesMap | undefined
+  >(undefined);
+  const [timelineVisEmbeddable, setTimelineVisEmbeddable] = useState<
+    VisualizeEmbeddable | undefined
+  >(undefined);
+  const [timeRange, setTimeRange] = useState<TimeRange | undefined>(undefined);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
+
+  function reload() {
+    visEmbeddable?.reload();
+    eventVisEmbeddablesMap?.forEach((embeddableItems) => {
+      embeddableItems.forEach((embeddableItem) => {
+        embeddableItem.embeddable.reload();
+      });
+    });
+  }
+
+  useEffect(() => {
+    fetchVisEmbeddable(props.savedObjectId, setTimeRange, setVisEmbeddable, setErrorMessage);
+    /* eslint-disable */
+  }, [props.savedObjectId]);
+
+  useEffect(() => {
+    if (visEmbeddable?.visLayers) {
+      createEventEmbeddables(
+        props.savedObjectId,
+        visEmbeddable,
+        setEventVisEmbeddablesMap,
+        setErrorMessage
+      );
+      createTimelineEmbeddable(
+        props.savedObjectId,
+        visEmbeddable,
+        setTimelineVisEmbeddable,
+        setErrorMessage
+      );
+    }
+  }, [visEmbeddable?.visLayers]);
+
+  useEffect(() => {
+    if (
+      visEmbeddable !== undefined &&
+      eventVisEmbeddablesMap !== undefined &&
+      timeRange !== undefined &&
+      timelineVisEmbeddable !== undefined
+    ) {
+      setIsLoading(false);
+    }
+  }, [visEmbeddable, eventVisEmbeddablesMap, timeRange, timelineVisEmbeddable]);
+
+  return (
+    <>
+      <EuiFlyout size="l" onClose={props.onClose}>
+        <EuiFlyoutHeader hasBorder>
+          <EuiTitle size="m">
+            <h1>{isLoading || errorMessage ? <>&nbsp;</> : `${visEmbeddable?.getTitle()}`}</h1>
```

Comment:
I had done that to preserve the flyout header height to being static, so that it doesn't change sizes when transitioning from loading -> loaded state. I'll switch to the text loading component and "Error fetching events" options so it's more informative.